include/deal.II/base/table.h

```diff
@@ -2367,6 +2624,177 @@ Table<2,T>::n_cols () const
 
 
 
+//---------------------------------------------------------------------------
+namespace TransposeTableIterators
+{
+  template <typename T, bool Constness>
+  inline
+  AccessorBase<T, Constness>::AccessorBase()
+    :
+    container(nullptr),
+    linear_index(std::numeric_limits<decltype(linear_index)>::max())
+  {}
+
+
+
+  template <typename T, bool Constness>
+  inline
+  AccessorBase<T, Constness>::AccessorBase(const container_pointer_type table)
+    :
+    container(table),
+    linear_index(container->values.size())
+  {}
+
+
+
+  template <typename T, bool Constness>
+  inline
+  AccessorBase<T, Constness>::AccessorBase(const AccessorBase<T, false> &a)
+    :
+    container(a.container),
+    linear_index(a.linear_index)
+  {}
+
+
+
+  template <typename T, bool Constness>
+  inline
+  AccessorBase<T, Constness>::AccessorBase(const container_pointer_type table,
+                                           const std::ptrdiff_t         index)
+    :
+    container(table),
+    linear_index(index)
+  {
+    Assert(0 <= linear_index && linear_index < container->values.size(),
+           ExcMessage("The current iterator points outside of the table."));
+  }
+
+
+
+  template <typename T, bool Constness>
+  inline
+  const T &
+  AccessorBase<T, Constness>::value() const
+  {
+    Assert(0 <= linear_index && linear_index < container->values.size(),
+           ExcMessage("The current iterator points outside of the table."));
+    return this->container->values[linear_index];
+  }
+
+
+
+  template <typename T, bool Constness>
+  inline
+  typename AccessorBase<T, Constness>::size_type
+  AccessorBase<T, Constness>::AccessorBase::row() const
+  {
+    Assert(!container->empty(), ExcMessage("An empty table has no rows or columns."));
+    const auto row_n = linear_index % container->n_rows();
+    Assert(0 <= row_n && row_n < container->m(),
```

Comment:
The failing tests are suggesting that this should be `container->n_rows()` (or `TransposeTable` should implement `m()`).